src/Controller/SentencesListsController.php

```diff
@@ -323,9 +324,18 @@ public function remove_sentence_from_list($sentenceId, $listId)
             $this->set('_serialize', ['removed']);
             $this->RequestHandler->renderAs($this, 'json');
         }
+                
+        if($origin == 0)
+        {
+            return $this->redirect([
+                'controller' => 'sentences',
+                'action' => 'show',
+                $sentenceId
+            ]);
+        }
```

Comment:
You don't need to add an additional parameter to find out where the request comes from. The request contains a `Referer` header with all the necessary information. In a controller the referer string is available through `$this->referer()`.

So what I've had in mind was 
````php
if (strpos($this->referer(), 'sentences/show')) {
    return $this->redirect($this->referer());
}
````

And I think you could also move the condition up just after the call to `$this->SentencesLists->removeSentenceFromList` because in the case where the request comes from `sentences/show` there's nothing else to do.